src/App.tsx

```diff
@@ -15,48 +16,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goodsFromServer];
```

Comment:
I like the fact that you are aware of mutating behavior of the Array methods you are using here and because of that you have created a copy of data.

"fun" fact => Recently we have `toSorted` metod available in JS which actually creates a shallow copy of array and returns sorted data without mutation of the array it was used on.